Author: Christoph Berg <myon@debian.org>
Description: Move libpgport/libpgcommon/libpgfeutils from libdir to pkglibdir
 This allows client applications to link to version-specific libraries.
 Used by pg-checksums.
Forwarded: No, (somewhat) Debian specific

Index: postgresql-12/src/common/Makefile
===================================================================
--- postgresql-12.orig/src/common/Makefile	2023-11-22 21:54:00.509883730 -0500
+++ postgresql-12/src/common/Makefile	2023-11-22 21:54:00.505883736 -0500
@@ -79,15 +79,15 @@
 
 # libpgcommon is needed by some contrib
 install: all installdirs
-	$(INSTALL_STLIB) libpgcommon.a '$(DESTDIR)$(libdir)/libpgcommon.a'
-	$(INSTALL_STLIB) libpgcommon_shlib.a '$(DESTDIR)$(libdir)/libpgcommon_shlib.a'
+	$(INSTALL_STLIB) libpgcommon.a '$(DESTDIR)$(pkglibdir)/libpgcommon.a'
+	$(INSTALL_STLIB) libpgcommon_shlib.a '$(DESTDIR)$(pkglibdir)/libpgcommon_shlib.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgcommon.a'
-	rm -f '$(DESTDIR)$(libdir)/libpgcommon_shlib.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgcommon.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgcommon_shlib.a'
 
 libpgcommon.a: $(OBJS_FRONTEND)
 	rm -f $@
Index: postgresql-12/src/fe_utils/Makefile
===================================================================
--- postgresql-12.orig/src/fe_utils/Makefile	2023-11-22 21:54:00.509883730 -0500
+++ postgresql-12/src/fe_utils/Makefile	2023-11-22 21:54:00.505883736 -0500
@@ -35,13 +35,13 @@
 
 # libpgfeutils could be useful to contrib, so install it
 install: all installdirs
-	$(INSTALL_STLIB) libpgfeutils.a '$(DESTDIR)$(libdir)/libpgfeutils.a'
+	$(INSTALL_STLIB) libpgfeutils.a '$(DESTDIR)$(pkglibdir)/libpgfeutils.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgfeutils.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgfeutils.a'
 
 clean distclean:
 	rm -f libpgfeutils.a $(OBJS) lex.backup
Index: postgresql-12/src/port/Makefile
===================================================================
--- postgresql-12.orig/src/port/Makefile	2023-11-22 21:54:00.509883730 -0500
+++ postgresql-12/src/port/Makefile	2023-11-22 21:54:00.505883736 -0500
@@ -50,15 +50,15 @@
 
 # libpgport is needed by some contrib
 install: all installdirs
-	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(libdir)/libpgport.a'
-	$(INSTALL_STLIB) libpgport_shlib.a '$(DESTDIR)$(libdir)/libpgport_shlib.a'
+	$(INSTALL_STLIB) libpgport.a '$(DESTDIR)$(pkglibdir)/libpgport.a'
+	$(INSTALL_STLIB) libpgport_shlib.a '$(DESTDIR)$(pkglibdir)/libpgport_shlib.a'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(libdir)'
+	$(MKDIR_P) '$(DESTDIR)$(pkglibdir)'
 
 uninstall:
-	rm -f '$(DESTDIR)$(libdir)/libpgport.a'
-	rm -f '$(DESTDIR)$(libdir)/libpgport_shlib.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgport.a'
+	rm -f '$(DESTDIR)$(pkglibdir)/libpgport_shlib.a'
 
 libpgport.a: $(OBJS)
 	rm -f $@
Index: postgresql-12/src/Makefile.global.in
===================================================================
--- postgresql-12.orig/src/Makefile.global.in	2023-11-22 21:54:00.509883730 -0500
+++ postgresql-12/src/Makefile.global.in	2023-11-22 21:55:43.621667841 -0500
@@ -585,7 +585,7 @@
 ifeq ($(PORTNAME),darwin)
 libpq_pgport = $(libpq)
 else ifdef PGXS
-libpq_pgport = -L$(libdir) -lpgcommon -lpgport $(libpq)
+libpq_pgport = -L$(pkglibdir) -lpgcommon -lpgport $(libpq)
 else
 libpq_pgport = -L$(top_builddir)/src/common -lpgcommon -L$(top_builddir)/src/port -lpgport $(libpq)
 endif
@@ -596,7 +596,7 @@
 # done if they don't, since they will have satisfied all their references
 # from these libraries.)
 ifdef PGXS
-libpq_pgport_shlib = -L$(libdir) -lpgcommon_shlib -lpgport_shlib $(libpq)
+libpq_pgport_shlib = -L$(pkglibdir) -lpgcommon_shlib -lpgport_shlib $(libpq)
 else
 libpq_pgport_shlib = -L$(top_builddir)/src/common -lpgcommon_shlib -L$(top_builddir)/src/port -lpgport_shlib $(libpq)
 endif
